packages/gestalt/src/Spinner/VRSpinner.tsx

```diff
@@ -6,40 +7,88 @@ import { useDefaultLabelContext } from '../contexts/DefaultLabelProvider';
 const SIZE_NAME_TO_PIXEL = {
   sm: 32,
   md: 40,
-  lg: 48,
+  lg: 56,
 } as const;
 
-type Props = {
-  accessibilityLabel?: string;
-  delay?: boolean;
+type SpinnerBodyProps = {
+  accessibilityLabel: string;
+  delay: boolean;
   show: boolean;
-  size?: 'sm' | 'md' | 'lg';
+  size: 'sm' | 'md' | 'lg';
+  onExitAnimationEnd: () => void;
 };
 
-export default function Spinner({ accessibilityLabel, delay = true, show, size = 'md' }: Props) {
-  const { accessibilityLabel: accessibilityLabelDefault } = useDefaultLabelContext('Spinner');
-
-  if (!show) return null;
-
+function SpinnerBody({
+  accessibilityLabel,
+  delay,
+  show,
+  size,
+  onExitAnimationEnd,
+}: SpinnerBodyProps) {
   return (
     <Box display="flex" justifyContent="around">
       <div
-        aria-label={accessibilityLabel ?? accessibilityLabelDefault}
-        className={classnames(styles.spinner, { [styles.delay]: delay })}
+        aria-label={accessibilityLabel}
+        className={classnames(styles.spinner, { [styles.exit]: !show })}
+        onAnimationEnd={onExitAnimationEnd}
         style={
           {
+            '--g-enter-delay': delay ? '300ms' : undefined,
             '--g-size': `${SIZE_NAME_TO_PIXEL[size]}px`,
           } as React.CSSProperties
         }
       >
-        <div className={styles.spinnerFrame}>
-          <div className={styles.dot1} />
-          <div className={styles.dot2} />
-          <div className={styles.dot3} />
-        </div>
+        <svg fill="none" viewBox="0 0 56 56" xmlns="http://www.w3.org/2000/svg">
+          <g style={{ '--g-start-x': '14px', '--g-start-y': '-8px' } as React.CSSProperties}>
+            <circle cx="14" cy="36" r="8" />
+          </g>
+          <g style={{ '--g-start-x': '0px', '--g-start-y': '16px' } as React.CSSProperties}>
+            <circle cx="28" cy="12" r="8" />
+          </g>
+          <g style={{ '--g-start-x': '-14px', '--g-start-y': '-8px' } as React.CSSProperties}>
+            <circle cx="42" cy="36" r="8" />
+          </g>
+        </svg>
       </div>
     </Box>
   );
 }
 
+type Props = {
+  accessibilityLabel?: string;
+  delay?: boolean;
+  show: boolean;
+  size?: 'sm' | 'md' | 'lg';
+};
+
+export default function Spinner({
+  accessibilityLabel,
+  delay = true,
+  show: showProp,
+  size = 'md',
+}: Props) {
+  const [show, setShow] = useState(showProp);
+  const { accessibilityLabel: accessibilityLabelDefault } = useDefaultLabelContext('Spinner');
+
+  const unmountSpinner = () => {
+    if (!showProp) setShow(false);
+  };
+
+  useEffect(() => {
+    if (showProp) setShow(showProp);
```

Comment:
should we do here something like...

 if (showProp !== show) setShow(showProp);

?